### PR TITLE
Promote is_bit_mask() function to common utils

### PR DESCRIPTION
**Pull request by soburi**
From the discussion #51955 (comment),
is_bit_mask() function can use commonly in whole system.

Move it to sys/util.h.

### CODE REVIEW

**Comment by soburi** [comment]
It will cause compile failure if both applied the https://github.com/zephyrproject-rtos/zephyr/pull/51955 and this,
I will rework after #51955 merged.

**Review by gmarull — DISMISSED** [review]

**Review by carlocaione — CHANGES_REQUESTED** [review]
This is a bit misleading. A bit mask can also be something like `0xff00`. So if we want to have this in utils we should either change name (or make it more general) and support 64-bit platforms as well.

**Comment by nordic-krch** [comment]
There is already `BIT_MASK()` macro in `util_macro.h`. What about converting it to macro `IS_BIT_MASK` and putting it next to `BIT_MASK`. That would kind of bind them together.
There is also `BIT64_MASK` so `IS_BIT64_MASK` could also be added.

**Comment by carlescufi** [comment]
> There is already `BIT_MASK()` macro in `util_macro.h`. What about converting it to macro `IS_BIT_MASK` and putting it next to `BIT_MASK`. That would kind of bind them together. There is also `BIT64_MASK` so `IS_BIT64_MASK` could also be added.

@soburi I think this is a good suggestion.

**Comment by soburi** [comment]
@nordic-krch @carlescufi 
Thank you for proposal.
I also think it is good. Applied it.

In this case, IS_BIT_MASK() is not dependent on bit-width of argument. No need to create IS_BIT64_MASK().

@carlocaione 

> This is a bit misleading. A bit mask can also be something like 0xff00. So if we want to have this in utils we should either change name (or make it more general) and support 64-bit platforms as well.

I also created IS_SHIFTED_BIT_MASK() that  can specify the start bit of the mask for support more generic case.
It is useful with using 'count trailing zeros' function.
To keep header dependency simple, I didn't implements wrapper macros.

File: include/zephyr/sys/util_macro.h [hunk 1]
```diff
@@ -73,6 +73,23 @@ extern "C" {
...
+ * @param s Specify the lowest bit for that is continuously set bits.
+ */
+#define IS_SHIFTED_BIT_MASK(m, s) (!(((m) >> (s)) & (((m) >> (s)) + 1U)))
+
+/**
+ * @brief Check if bits are set continuously from the LSB.
+ *
+ * @param m Check whether the bits are set continuously from LSB.
+ */
+#define IS_BIT_MASK(m) IS_SHIFTED_BIT_MASK(m, 0)
```
Comment (gmarull) [inline include/zephyr/sys/util_macro.h:91]: tests need to be added

Comment (soburi) [reply include/zephyr/sys/util_macro.h:91]: I added tests to `tests/unit/util`.

File: tests/unit/util/test.inc [hunk 2]
```diff
@@ -526,3 +526,89 @@ void run_ARRAY_INDEX_FLOOR(void)
...
+	uint32_t bitmask32_shift1 = 0xfffffffeUL;
+	uint32_t bitmask32_shift31 = 0x80000000UL;
+	uint64_t bitmask64_shift1 =  0xfffffffffffffffeULL;
+	uint64_t bitmask64_shift63 = 0x8000000000000000ULL;
+
+	zassert_true(IS_SHIFTED_BIT_MASK(bitmask32_shift1, 1));
+	zassert_true(IS_SHIFTED_BIT_MASK(bitmask32_shift31, 31));
+	zassert_true(IS_SHIFTED_BIT_MASK(bitmask64_shift1, 1));
+	zassert_true(IS_SHIFTED_BIT_MASK(bitmask64_shift63, 63));
+	//zassert_false(IS_SHIFTED_BIT_MASK(bitmask32_shift31, 32));
```
Comment (nordic-krch) [inline tests/unit/util/test.inc:605]: please uncomment or remove.

File: tests/unit/util/test.inc [hunk 3]
```diff
@@ -526,3 +526,89 @@ void run_ARRAY_INDEX_FLOOR(void)
...
+	//zassert_false(IS_SHIFTED_BIT_MASK(bitmask64_shift63, 64));
+
+	zassert_true(IS_SHIFTED_BIT_MASK(0xfffffffeUL, 1));
+	zassert_true(IS_SHIFTED_BIT_MASK(0xfffffffffffffffeULL, 1));
+	zassert_true(IS_SHIFTED_BIT_MASK(0x80000000UL, 31));
+	zassert_true(IS_SHIFTED_BIT_MASK(0x8000000000000000ULL, 63));
+	//zassert_false(IS_SHIFTED_BIT_MASK(0x800000000UL, 32));
```
Comment (nordic-krch) [inline tests/unit/util/test.inc:612]: Ditto

File: tests/unit/util/test.inc (hunk 2 above)
Comment (soburi) [reply tests/unit/util/test.inc]: Sorry. I forgot remove it.
Fixed it.

**Review by nordic-krch — DISMISSED** [review]

File: tests/unit/util/test.inc [hunk 4]
```diff
@@ -526,3 +526,85 @@ void run_ARRAY_INDEX_FLOOR(void)
...
+void run_IS_BIT_MASK(void)
+{
+	uint32_t zero32 = 0UL;
+	uint64_t zero64 = 0ULL;
+	uint32_t bitmask1 = 0x00000001UL;
+	uint32_t bitmask2 = 0x00000003UL;
+	uint32_t bitmask31 = 0x7fffffffUL;
+	uint32_t bitmask32 = 0xffffffffUL;
+	uint64_t bitmask63 = 0x7fffffffffffffffUL;
+	uint64_t bitmask64 = 0xffffffffffffffffUL;
```
Comment (carlocaione) [inline tests/unit/util/test.inc:565]: `ULL`?

**Review by carlocaione — CHANGES_REQUESTED** [review]

File: tests/unit/util/test.inc (hunk 4 above)
Comment (soburi) [reply tests/unit/util/test.inc]: I overlooked it. Thank you
Fixed it.

**Review by carlocaione — APPROVED** [review]

**Review by nordic-krch — APPROVED** [review]

**Review by nandojve — APPROVED** [review]

**Pull request merged**